Date: Oct 21, 2022
From:
Subject: Process WPS_CANCEL while in authenticating/associating state also
Source: COMCAST
License:
Upstream-Status: Pending
Signed-off-by: Tony Ukken <Tony_Ukken2@cable.comcast.com>

Index: wpa_supplicant/wps_supplicant.c
===================================================================
--- a/wpa_supplicant/wps_supplicant.c
+++ b/wpa_supplicant/wps_supplicant.c
@@ -1336,7 +1336,7 @@ int wpas_wps_cancel(struct wpa_supplican
 		wpa_printf(MSG_DEBUG, "WPS: Cancel operation - cancel scan");
 		wpa_supplicant_cancel_scan(wpa_s);
 		wpas_clear_wps(wpa_s);
-	} else if (wpa_s->wpa_state >= WPA_ASSOCIATED) {
+	} else if (wpa_s->wpa_state >= WPA_AUTHENTICATING) {
 		wpa_printf(MSG_DEBUG, "WPS: Cancel operation - "
 			   "deauthenticate");
 		wpa_s->own_disconnect_req = 1;
