From 6fd58d4d971c1437b4c5740cc2903899f23cf3fc Mon Sep 17 00:00:00 2001
From: djalad690_comcast <Devika_Jaladi@comcast.com>
Date: Thu, 23 Jan 2025 12:46:00 +0000
Subject: [PATCH] wpa_supplicant patch "fix_HS20_build_with_MBO.patch" need to
 be corrected

---
 wpa_supplicant/interworking.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/wpa_supplicant/interworking.c b/wpa_supplicant/interworking.c
index 71a5c16..59fa9b6 100644
--- a/wpa_supplicant/interworking.c
+++ b/wpa_supplicant/interworking.c
@@ -1620,7 +1620,7 @@ static int interworking_connect_roaming_consortium(
 
 	if (interworking_set_hs20_params(wpa_s, ssid) < 0)
 		goto fail;
-
+#ifdef CONFIG_HS20
 	ie = wpa_bss_get_ie(bss, WLAN_EID_ROAMING_CONSORTIUM);
 	anqp = bss->anqp ? bss->anqp->roaming_consortium : NULL;
 	for (i = 0; (ie || anqp) && i < cred->num_roaming_consortiums; i++) {
@@ -1640,7 +1640,7 @@ static int interworking_connect_roaming_consortium(
 			cred->roaming_consortiums_len[i];
 		break;
 	}
-
+#endif /* CONFIG_HS20 */
 	if (cred->eap_method == NULL) {
 		wpa_msg(wpa_s, MSG_DEBUG,
 			"Interworking: No EAP method set for credential using roaming consortium");
